### PR TITLE
integrate sphinx

### PR DESCRIPTION
**Pull request by sam-goldman**
Integrates Sphinx into the deployment process.

## Notes

* We support all of the networks that you deploy on, including Zora and RARI, which we added after our recent call.
* I needed to make a couple slight changes to the `BaseChainSetup` contract in the `forge-toolkit` repo, so I forked that repo and use it as a dependency here. The diff of the changes is [here](https://github.com/sphinx-labs/forge-toolkit/commit/226cce3c06c948103c00824a5d9f033a30c52b72).
* If you have any questions, check out our [GitHub page](https://github.com/sphinx-labs/sphinx) or send us a message.

## Deployment Instructions

Follow these steps to deploy with Sphinx on a few testnets. It should take you 5 minutes to finish.

1. Clone this fork of your repo:
```
git clone git@github.com:sphinx-labs/decent-bridge.git
```

2.
```
cd decent-bridge
```

3. Install packages:
```
pnpm install && pnpm sphinx install
```

4. Sign up for Sphinx using this [invite link](https://www.sphinx.dev/signup?code=4e81ac8b-0b4f-41f5-aa9c-e0e1241a92ff).

5. In Sphinx's website, go to "Options" -> "API Credentials". You'll need these credentials in the next couple of steps.

6. Open `script/Scripts.s.sol`. The `setUp` function contains your config options. Update the following fields:\
    a. In `sphinxConfig.orgId`, add the Organization ID from Sphinx's website. This is a public field, so you don't need to keep it secret.\
    b. In `sphinxConfig.owners`, add the addresses of the account(s) that will own your project. (Specifically, they'll own the Gnosis Safe that executes your deployment).

7. Create a `.env` file:
```
cp .env.example .env
```

8. Open the `.env`, then enter values for the `SPHINX_API_KEY` as well as the RPC endpoints. The `SPHINX_API_KEY` is in the Sphinx UI (under "Options" -> "API Credentials").

9. You're done with the configuration steps! Run `forge test` to make sure your tests are passing.

10. Next, propose the script called `Deploy` on the networks in `sphinxConfig.testnets`:
```
pnpm sphinx propose script/Scripts.s.sol --networks testnets --target-contract Deploy
```

11. When the proposal is finished, go to the [Sphinx UI](https://sphinx.dev) to approve the deployment. After you approve it, you can monitor the deployment's status in the UI while it's executed.

12. (Optional) If you'd like to propose any of the other scripts, just add the `sphinx` modifier to its `run` function. Then, run the same `propose` command in step 10 targeting the new script (via `--target-contract <CONTRACT_NAME>`).

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -21,4 +21,7 @@
 	url = https://github.com/decentxyz/better-deployer
 [submodule "lib/forge-toolkit"]
 	path = lib/forge-toolkit
-	url = https://github.com/decentxyz/forge-toolkit
+	url = https://github.com/sphinx-labs/forge-toolkit
```
Comment (sam-goldman) [inline .gitmodules:24]: Changed this URL to be my fork of the `forge-toolkit` repo

File: foundry.toml [hunk 2]
```diff
@@ -8,22 +8,26 @@ fs_permissions = [
...
 [rpc_endpoints]
 ethereum = "${ETHEREUM_RPC}"
 arbitrum = "${ARBITRUM_RPC}"
 optimism = "${OPTIMISM_RPC}"
 polygon = "${POLYGON_RPC}"
 avalanche = "${AVALANCHE_RPC}"
-ftm-testnet = "${FTM_TESTNET_RPC}"
-fuji = "${FUJI_RPC}"
+fantom_testnet = "${FTM_TESTNET_RPC}"
+avalanche_fuji = "${FUJI_RPC}"
```
Comment (sam-goldman) [inline foundry.toml:25]: Changed these RPC endpoint names because Sphinx expects that they have specific names

File: foundry.toml [hunk 3]
```diff
@@ -8,22 +8,26 @@ fs_permissions = [
...
 sepolia = "${SEPOLIA_RPC}"
 zora = "${ZORA_RPC}"
 base = "${BASE_RPC}"
-zora-goerli = "${ZORA_GOERLI_RPC}"
-optimism-goerli = "${OPTIMISM_GOERLI_RPC}"
+zora_sepolia = "${ZORA_SEPOLIA_RPC}"
+optimism_sepolia = "${OPTIMISM_SEPOLIA_RPC}"
```
Comment (sam-goldman) [inline foundry.toml:30]: Upgraded these to Sepolia because Goerli is deprecated

**Review by RPate97 — APPROVED** [review]
Looks good, there's just one issue with the env variable for optimism sepolia not being specified in the example env file.

File: .env.example [hunk 4]
```diff
@@ -2,6 +2,9 @@
 TESTNET_ACCOUNT=
 MAINNET_ACCOUNT=
 
+#### sphinx
+SPHINX_API_KEY=
+
```
Comment (RPate97) [inline .env.example:7]: The script calls for deploying on `optimism_sepolia`, but this file has not been updated to include an RPC URL for that network.

**Pull request merged**